opgee/xml_utils.py

```diff
@@ -59,6 +79,10 @@ def attr_to_xml(fields, dtypes, xml_path, analysis_name, modifies='default'):
     group = ET.SubElement(analysis, "Group")
     group.text = "all"
 
+    # we use this to look up enclosing <Aggregator> nodes for <Process> nodes
+    template_field = _load_opgee_template(modifies)
+    aggregators = template_field.findall("Aggregator")
```

Comment:
Not quite sure what you're asking. This was to ensure that when XML is merged with the "template", the correct nesting is used. Otherwise there were multiple instances of the same named process, one connected properly and another (perhaps with user-defined attribute values) that was not connected. Implementing the validation system unearthed this.

Reply:
Ah, I see. My initial reaction was that this change made it look like we weren't instantiating the `Aggregator` objects defined in the XML at all. Good to know that's not the case.